test/unit_test.c

```diff
@@ -2111,17 +2111,17 @@ static const char *test_coap(void) {
     0x65, 0x73, 0x6f, 0x75, 0x72, 0x63, 0x65, 0xf1,
     0x6d, 0x79, 0x64, 0x61, 0x74, 0x61 };
 
-  memset(&io, 0, sizeof(io));
-
+  iobuf_init(&packet_in, 0);
   /* empty buf */
-  res = coap_parse(&io, &cm);
+  res = coap_parse(&packet_in, &cm);
   ASSERT((res & NS_COAP_NOT_ENOUGH_DATA) != 0);
   ns_coap_free_options(&cm);
 
+  iobuf_init(&packet_out, 0);
   /* ACK, MID: 59675, Empty Message */
-  io.buf = coap_packet_2;
-  io.len = sizeof(coap_packet_2);
-  res = coap_parse(&io, &cm);
+  packet_in.buf = coap_packet_2;
```

Comment:
minor:

leakiness depends on implementation detail of iobufs:

iobufs are initialized with a NULL buf pointer and allocation happens only on actual growth. However, if this behaviour is changed at any time (possibly optionally depending on some compilation time flag), here we leak a memory block.

This might cause spurious leak analysis regressions that introduce noise and reduce effectiveness of the build automation tooling.

---

Just put a `iobuf_free(&packet_in)` call at the end of the previous test (L2119).